core/node/consensus/src/storage/store.rs

```diff
@@ -203,93 +215,130 @@ impl StoreRunner {
                 const POLL_INTERVAL: time::Duration = time::Duration::seconds(1);
                 let mut next_batch_number = { batches_persisted.borrow().next() };
                 loop {
-                    let mut conn = pool.connection(ctx).await?;
-                    if let Some(last_batch_number) = conn
-                        .get_last_batch_number(ctx)
-                        .await
-                        .wrap("last_batch_number()")?
-                    {
-                        if last_batch_number >= next_batch_number {
-                            let range = conn.batches_range(ctx).await.wrap("batches_range()")?;
-                            next_batch_number = last_batch_number.next();
-                            batches_persisted.send_replace(range);
+                    let span = tracing::info_span!("gossip_sync_batches_loop", %next_batch_number);
+                    async {
+                        let mut conn = pool.connection(ctx).await?;
+                        if let Some(last_batch_number) = conn
+                            .get_last_batch_number(ctx)
+                            .await
+                            .wrap("last_batch_number()")?
+                        {
+                            if last_batch_number >= next_batch_number {
+                                let range =
+                                    conn.batches_range(ctx).await.wrap("batches_range()")?;
+                                next_batch_number = last_batch_number.next();
+                                tracing::info_span!("batches_persisted_send").in_scope(|| {
+                                    batches_persisted.send_replace(range);
+                                });
+                            }
                         }
+                        ctx.sleep(POLL_INTERVAL)
+                            .instrument(tracing::info_span!("sleep"))
+                            .await?;
+
+                        anyhow::Ok(())
                     }
-                    ctx.sleep(POLL_INTERVAL).await?;
+                    .instrument(span)
+                    .await?;
                 }
             });
 
             s.spawn::<()>(async {
                 // Loop inserting batch certificates into storage
                 const POLL_INTERVAL: time::Duration = time::Duration::milliseconds(50);
                 loop {
-                    let cert = batch_certificates.recv(ctx).await?;
+                    let span = tracing::info_span!("insert_batch_certificates_loop");
+
+                    async {
+                        let cert = batch_certificates
+                            .recv(ctx)
+                            .instrument(tracing::info_span!("wait_for_certificate"))
+                            .await?;
+
+                        loop {
```

Comment:
any reason you switched the order of the tasks?

Reply:
I didn't change the order of the tasks. Github just generated a weird diff because of the indentation changes.